File: CDFC/1500/E_Special_Elements.py
"""
==========================================================
|  Archivo       : E_Special_Elements.py
|  Autor         : Russell
|  Fecha         : 2026-02-07 17:23
|--------------------------------------------------------
|  Tópicos utilizados:
|  - number theory
|  - training
|  - Lev 2
==========================================================
"""

from typing import List,Dict
from collections import defaultdict
import sys

def gcd(a: int, b: int) -> int:
    """Máximo común divisor"""
    if a % b == 0:
        return b
    return gcd(b, a % b)

def floordiv(a: int, b: int) -> int:
    """División entera hacia abajo (estilo C++)"""
    if a >= 0:
        return a // b
    return -((-a + b - 1) // b)

def ceildiv(a: int, b: int) -> int:
    """División entera hacia arriba"""
    if a >= 0:
        return (a + b - 1) // b
    return -((-a) // b)

def bitlen_unsigned(n: int) -> int:
    """Equivalente a __builtin_clzll"""
    return 1 if n == 0 else n.bit_length()


def solve() -> None:
    n: int = int(sys.stdin.readline())
    a: List[int] = list(map(int, sys.stdin.readline().split()))

    prefix_sum: Dict[int, bool] = {}

    for i in range(n - 1):
        s: int = a[i]
        for j in range(i + 1, n):
            s += a[j]
            if s >= n + 1:
                break
            prefix_sum[s] = True

    ans: int = 0
    for ai in a:
        if ai in prefix_sum:
            ans += 1

    print(ans)


def main() -> None:
    # CASO MULTI-TEST
    t = int(sys.stdin.readline())
    for _ in range(t):
        solve()

    # CASO SINGLE-TEST
    # solve()

if __name__ == "__main__":
    main()

File: CDFC/1500/E_Special_Elements.cpp
/*
==========================================================
|  Archivo       : E_Special_Elements.cpp
|  Autor         : Russell
|  Fecha         : 2026-02-07 12:31
|--------------------------------------------------------
|  Tópicos utilizados:
|  - brute force
|  - implementation
|  - Data structure: unordered_map
|  - *1500
|  - Lev 2
==========================================================
*/

#include <bits/stdc++.h>
#define all(x) x.begin(),x.end()
using namespace std;
using ll = long long;
using i64 = int64_t;
using i32 = int32_t;
using ld = long double;
using uint = unsigned int;
using ull = unsigned long long;
template<typename T>
using pair2 = pair<T, T>;
using pii = pair<int, int>;
using pli = pair<ll, int>;
using pll = pair<ll, ll>;
using vll = vector<ll>;

#define pb push_back
#define mp make_pair

const int INF = 1e9;

ll gcd(ll a, ll b) {
    if (a % b == 0) return b;
    else return gcd(b, a % b);
}

ll floordiv(ll a, ll b) {
    if (a >= 0) return a / b;
    return - ( (-a + b - 1) / b );
}

ll ceildiv(ll a, ll b) {
    // ceil(a/b) = -floor((-a)/b)
    if (a >= 0) return (a + b - 1) / b;
    return - ( (-a) / b );
}

int bitlen_unsigned(unsigned long long n) {
    // __builtin_clzll: Cuenta los ceros a la izquierda
    return (n == 0) ? 1 : 64 - __builtin_clzll(n);
}

clock_t startTime;
double getCurrentTime() {
    return (double)(clock() - startTime) / CLOCKS_PER_SEC;
}

void solve() {
    int n; cin >> n;
    vector<int> a(n);
    for(int i = 0; i < n; i++) cin >> a[i];
    unordered_map<int, bool> prefixSum;
    for(int i = 0; i < n - 1; i++){
        int sum = a[i];
        for(int j =  i+1; j < n; j++){
            sum += a[j];
            if(sum >= n + 1) break;
            prefixSum[sum] = true;
        }
    }
    int ans = 0;
    for(auto ai: a){
        if(prefixSum.count(ai) > 0) ans++;
    }
    cout << ans << endl;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    #ifdef DEBUG
        freopen("input.txt", "r", stdin);
        freopen("output.txt", "w", stdout);
    #endif

    int t;
    cin >> t;
    // t = 1;
    while (t--) {
        solve();
    }
    return 0;
}